crates/ruff_python_parser/src/parser/expression.rs

```diff
@@ -2777,18 +2784,28 @@ impl<'src> Parser<'src> {
             unreachable!()
         };
 
-        if !matches!(kind, IpyEscapeKind::Magic | IpyEscapeKind::Shell) {
-            // This should never occur as the lexer won't allow it.
-            unreachable!("IPython escape command expression is only allowed for % and !");
-        }
-
```

Comment:
I think this needs to be reverted now that the parser correctly recovers from the syntax error